examples/regression/plot_kim2020_simulations.py

```diff
@@ -0,0 +1,484 @@
+
+
+def B_random_from_B_fixed(
+    B: int, train_size: int, m: int, itrial: int = 0, random_state: int = 98765
+) -> int:
+    """
+    Generates a random number from a binomial distribution.
+
+    Parameters
+    ----------
+    B : int
+        Fixed B, parameter of the binomial distribution
+    train_size : int
+        Training set size
+    m : int
+        Resampling set size
+    itrial : int
+        Number of the trial
+    random_state
+        Base random state (fixed according to [1].)
+
+    Returns
+    -------
+    int
+        Integer drawn according to
+        Binomial(B/(1-1/(train_size+1))^m),(1-1/(train_size+1))^m),
+        where B, train_size and m are parameters.
+    """
+    np.random.seed(random_state + itrial)
+    return int(
+        np.random.binomial(
+            int(B / (1 - 1.0 / (1 + train_size)) ** m),
+            (1 - 1.0 / (1 + train_size)) ** m,
+            size=1,
+        )
+    )
+
+
+def comparison_JAB(
+    model: BaseEstimator = Ridge2(),
+    agg_function: str = "mean",
+    alpha: float = 0.1,
+    trials: int = 10,
+    train_size: int = 200,
+    boostrap_size: int = 10,
+    B_fixed: int = 50,
+    random_state: int = 98765,
+) -> pd.DataFrame:
+    """
+    Launch trials of jackknife-plus and jackknife-plus_after_boostrap,
+    with B fixed and random, for a given number of resample size and a given
+    number of trials, and returns the results as a DataFrame,
+
+    Parameters
+    ----------
+    model : BaseEstimator
+        Base model. By default, Ridge2.
+    agg_function: str
+        Aggregation function to test.
+    alpha : float
+        1 - (target coverage level).
+    trials: int
+        Number of trials launch for a given boostrap set size.
+    train_size : int
+        Size of the train set.
+    bootstrap_size : int
+        Number of boostrap sizes to test,
+        uniformly distributed between 10 and 100%
+        of the train set size.
+    B_fixed : int
+        Number of bootstrap samples in J+aB is drawn as
+        B ~ Binomial(int(B_fixed/(1-1/(n+1))^m),(1-1/(n+1))^m),
+        where n is the training set size, and m the resampling set size.
+    random_state : int
+        Random state. By default, 98765 (from [1]).
+
+    Returns
+    -------
+    pd.DataFrame
+        DataFrame with columns:
+        - itrial : the number of the trial
+        - model : the estimator's name
+        - method : jackknife+ of jackknife+-after-bootsrap
+        - coverage : PIs' coverage
+        - width : mean PI's width
+        - m : the resampling set size
+        - agg_function: aggregation method
+    """
+
+    results = pd.DataFrame(
+        columns=["itrial", "estimator", "method", "coverage", "width", "m"],
+        index=np.arange(trials * (2 * boostrap_size + 1)),
+    )
+
+    (X, y) = get_X_y()
+    m_vals = np.round(
+        train_size * np.linspace(0.1, 1, num=boostrap_size)
+    ).astype(int)
+    result_index = 0
+    gc.collect()
```

Comment:
Could you explain what this line does ?

Reply:
gc.collect() forces the memory of deleted object to be free
But it is no longer relevant since only one data set is tested
I delete it